### ISSUE TITLE
Не работает youtube.com на WebOS

### ISSUE THREAD

**Issue by Jenstel** [hufrea/byedpi#112]
Настроил это решение по гайду https://habr.com/ru/articles/838452/ на микротике. На компе и на телефонах, а также на телевизоре с Android все работает прекрасно. А вот на двух телеках LG приложение ютуб просто не запускается, пишет либо не удалось найти адрес youtube.com, либо отсутствует сетевое подключение. Но стоит только завернуть запрос к youtube.com (только на этот адрес) на ВПН, как приложение тут же запускается и дальше работает нормально уже давно с выключенным ВПНом. После закрытия приложения все повторяется. Помогите, в чем может быть проблема?

**Comment by Jenstel** [comment]
![image](https://github.com/user-attachments/assets/9d25000f-2dc7-470f-8f7b-61a73ff9f057)

**Comment by Jenstel** [comment]
Прошу прощения, проблема в домене www.youtube.com, именно если запрос на него сначала отправить в ВПН, то приложение начинает работать

**Comment by vanes32** [comment]
> Прошу прощения, проблема в домене [www.youtube.com](http://www.youtube.com), именно если запрос на него сначала отправить в ВПН, то приложение начинает работать

У меня с WebOS такая история:
На пк и телефоне ютюб летает с такой стратегией:
`--fake -1+s --ttl 1`
Телевизор с такой стратегией открывает приложение ютюба 1 раз из 50ти.
Методом тыка выяснил что проблема в домене youtube.com. Сделал для этого домена отдельную стратегию:
`--fake 8 --ttl 5 --hosts :youtube.com --auto=none --fake -1+s --ttl 1` 
Приложение на тв стало открываться, но медленно, может пару минут тупить. Продолжаю подбирать стратегию именно под этот домен на ТВ.

**Comment by AyOhEff** [comment]
Да, на WebOS (LG OLED) не работает простая настройка ByeDPI, может из-за QUIC?

**Comment by wiktorbgu** [comment]
Нашел для тестов телевизор samsung с tizen os для теста ютуба по моей инструкции [https://habr.com/ru/articles/838452/](https://habr.com/ru/articles/838452/)
Модель Samsung UE43AU7570UXRU обновил ПО в нем через меню на всякий случай.
Версия приложения ютуба web_20240908_19RC00
Оставил только такие хосты, а именно убрал [youtube.com](http://www.youtube.com/)
![изображение](https://github.com/user-attachments/assets/e7641a0d-c0df-4164-82b0-9daffd31f847)
При открытии приложения долго прогружается первые страницы разделов, но с включенным заворотом [youtube.com](http://www.youtube.com/) приложение вообще не запускается
Если перезагрузить роутер и ТВ, то высока вероятность что заработает. В моем случае работает, 4к летает
![изображение](https://github.com/user-attachments/assets/32b3cb28-8ef0-4dc0-80cd-a1235a6c61d9)

**Comment by tweaker** [comment]
> Если перезагрузить роутер и ТВ, то высока вероятность что заработает. В моем случае работает, 4к летает
Работает на ТВ (webos) и appletv, но страницы прогружаются от 15 сек до минуты, видео быстро грузится и в максимальном разрешении, но неюзабельно с такими тормозами. Пока оставил все через впн. Провайдер РТ.

**Comment by MrRanMao** [comment]
> > Если перезагрузить роутер и ТВ, то высока вероятность что заработает. В моем случае работает, 4к летает
> 
> Работает на ТВ (webos) и appletv, но страницы прогружаются от 15 сек до минуты, видео быстро грузится и в максимальном разрешении, но неюзабельно с такими тормозами. Пока оставил все через впн. Провайдер РТ.

У меня тоже РТ, советую еще поиграться с настройками ciadpi
У меня работает со следующей конфигурацией: `ciadpi --port 1234 --fake 1 --split 2 --ttl 2 --mod-http=dcsmix --auto=none`

Но как я ранее указал в другом [issue](https://github.com/hufrea/byedpi/issues/135), могут быть вылеты

**Comment by wiktorbgu** [comment]
> заблокирована

Доступна через vpn. 
Либо вот [https://web.archive.org/web/20240917203256/https://habr.com/ru/articles/838452/](https://web.archive.org/web/20240917203256/https://habr.com/ru/articles/838452/)

**Comment by OvalII** [comment]
у меня на LG TV заработала конфигурация из https://github.com/hufrea/byedpi/issues/236

**Comment by mgotch** [comment]
Тоже на WebOS не хотел работать. Начал перебирать все стратегии подряд iPad, телефон, браузер - без проблем.
Заработало  -Kt,h -s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1+s -s-1 -An -Ku -a5 -s443+s -d80+s -d443+s -s80+s -An

Полная конфигурация Debian, через которое ходит LG :
./ciadpi-i686 --transparent -p 12345 -Kt,h -s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1+s -s-1 -An -Ku -a5 -s443+s -d80+s -d443+s -s80+s -An

sudo iptables -A FORWARD -i enp3s0 -o enp3s0 -j ACCEPT
sudo iptables -t nat -A POSTROUTING -o enp3s0 -j MASQUERADE
sudo iptables -t nat -N REDSOCKS
sudo iptables -t nat -A REDSOCKS -p tcp --dport 443 -j REDIRECT --to-ports 12345
sudo iptables -t nat -A PREROUTING -i enp3s0 -p tcp -j REDSOCKS

**Comment by Jenstel** [comment]
После обновления на 16ую версию перестало работать на WebOS полностью, любая стратегия

**Comment by funkerwolf** [comment]
На последней версии работает через несколько попыток повтора подключений с телика.

> После обновления на 16ую версию перестало работать на WebOS полностью, любая стратегия

**Comment by Jenstel** [comment]
А какая стратегия?

**Comment by funkerwolf** [comment]
На днях обновился ютаб на теликах сам, теперь все РАБотает как положенно. Как понять что новая ерсия залетела у вас - спросит вы гость или кто на запуске.

**Comment by Jenstel** [comment]
Проблема с ТВ решается путем уменьшения MSS до 88 для запросов к www.youtube.com

**Comment by DeniR** [comment]
[Jenstel](https://github.com/Jenstel), и как уменишь mss до 88, какой пример конфига ? Гугл говорит, что параметр mss - это zapret, а не byedpi.

Текущий рабочий конфиг для byedpi под РТ (кроме WebOS) --hosts /etc/config/byedpi.hosts -Kt,h -s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1 +s -s-1 -An -b+500 --tls-sni=www.google.com --ttl 3 -Ku -a5 -s443+s -d80+s -d443+s -s80+s -An

**Comment by wiktorbgu** [comment]
@DeniR mangle = change mss = 88

**Comment by rfyodorov** [comment]
Поставил на роутер nfqws-keenetic (ушёл c byedpi)

рПровайдер Ростелеком, на всех домашних устройствах youtube не "тормозит" (включая родное приложение в LG WebOS) 

конфиг nfqws:
```
NFQWS_ARGS="--dpi-desync=fake,multisplit --dpi-desync-split-pos=1,midsld --dpi-desync-ttl=0 --dpi-desync-repeats=16 --dpi-desync-fooling=badseq,md5sig --dpi-desync-fake-tls=/opt/etc/nfqws/tls_clienthello_www_google_com.bin"

NFQWS_ARGS_UDP="--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=6 --dpi-desync-cutoff=n2"
```

**Comment by DeniR** [comment]
> [@DeniR](https://github.com/DeniR) mangle = change mss = 88

У меня openwrt, а не микротик. Куда это ?

**Comment by DeniR** [comment]
> Поставил на роутер nfqws-keenetic (ушёл c byedpi)

В zapret тоже есть nfqws, можно его использовать с теми же параметрами ?
Список хостов не используется, правила применяются абсолютно для всех сайтов ?

**Comment by rfyodorov** [comment]
> > Поставил на роутер nfqws-keenetic (ушёл c byedpi)
> 
> В zapret тоже есть nfqws, можно его использовать с теми же параметрами ? Список хостов не используется, правила применяются абсолютно для всех сайтов ?

Да, раньше использовал zapret с этими правилами. Но это работает с РТ, с другими провайдерами возможно потребуются использовать другие параметры.

правила применяются только к youtube, вот пример настройки за это отвечающиё
```
# auto - automatically detects blocked resources and adds them to the auto.list
NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"
```

user.list:
`youtube.com
youtu.be
googleapis.com
googlevideo.com
i.ytimg.com
i9.ytimg.com
yt3.ggpht.com
yt3.googleusercontent.com
yt4.ggpht.com
yt4.googleusercontent.com
gvt1.com
gstatic.com
youtube-ui.l.google.com
ytimg.l.google.com
ytstatic.l.google.com`

**Comment by DeniR** [comment]
> Да, раньше использовал zapret с этими правилами.

Чем nfqws-keenetic лучше zapret ?

**Comment by rfyodorov** [comment]
> > Да, раньше использовал zapret с этими правилами.
> 
> Чем nfqws-keenetic лучше zapret ?

nfqws-keenetic более лёгкий в нём только сам nfqws (а мне нужен только он) + можно установить вэб интерфейс
zapret более функциональный nfqws+tpws+ip2net+mdig + вспомогательные скрипты

**Comment by DeniR** [comment]
> nfqws-keenetic более лёгкий в нём только сам nfqws (а мне нужен только он) + можно установить вэб интерфейс

Спасибо, заработал youtube на LG WebOS ! Пришлось в openwrt в разделе wirewall отключить hardware flow offloading.
Только не пойму какой логин и пароль для веб-интерфейса, у меня же не keenetic, а openwrt на redmi ax6000.

**Comment by rfyodorov** [comment]
> > nfqws-keenetic более лёгкий в нём только сам nfqws (а мне нужен только он) + можно установить вэб интерфейс
> 
> Спасибо, заработал youtube на LG WebOS ! Пришлось в openwrt в разделе wirewall отключить hardware flow offloading. Только не пойму какой логин и пароль для веб-интерфейса, у меня же не keenetic, а openwrt на redmi ax6000.

Логин скорей всего root (и его пароль, под которым Вы по ssh подключаетесь)
Скрипт вэб интерфейса сравнивает логин\пароль с файлом /opt/etc/passwd

**Comment by funkerwolf** [comment]
> > На днях сменили схему блокировки в Ростелекоме, какой теперь рабочий конфиг для YT ?
> 
> Та же петрушка. В июне отрыгнуло, 3 месяц борюсь. Всё работает, кроме webos yt. Не нашли решение? На zapret сидел на openwrt, перешел на byedpi. Всё так же как и на zapret.

/ip firewall mangle add action=mark-connection chain=prerouting connection-mark=no-mark dst-address-list=za_dpi_FWD in-interface-list=LAN new-connection-mark=to_dpi passthrough=yes
/ip firewall mangle add action=mark-connection chain=prerouting connection-mark=no-mark dst-address-list=za_dpi_MSS_FWD in-interface-list=LAN new-connection-mark=to_dpi passthrough=yes
/ip firewall mangle add action=mark-routing chain=prerouting comment="To DPI" connection-mark=to_dpi in-interface-list=LAN new-routing-mark=dpi_mark passthrough=no routing-mark=!dpi_mark
/ip firewall mangle add action=change-mss chain=forward comment="Youtube MSS" dst-address-list=za_dpi_MSS_FWD new-mss=88 passthrough=no protocol=tcp src-address=192.168.254.2 tcp-flags=syn

**Comment by Vovanni** [comment]
Есть нестандартное решение которое мне помогло, если у вас также приложение для webos через byedpi не работает, но на остальных клиентах (браузер, андроид) все ок.
Я поставил клиент YouTube через Media Station X, он прекрасно работает через прокси.
